solver/result/result.go

```diff
@@ -185,3 +194,14 @@ func ConvertResult[U comparable, V comparable](r *Result[U], fn func(U) (V, erro
 
 	return r2, nil
 }
+
+func cloneMap[K comparable, V any](m map[K]V) map[K]V {
```

Comment:
Probably https://pkg.go.dev/golang.org/x/exp/maps#Clone can be used

Reply:
Switched to using this :tada: